django/db/models/functions/comparison.py

```diff
@@ -87,3 +87,14 @@ def __init__(self, *expressions, **extra):
     def as_sqlite(self, compiler, connection):
         """Use the MIN function on SQLite."""
         return super().as_sqlite(compiler, connection, function='MIN')
+
+
+class NullIf(Func):
+    function = 'NULLIF'
+    arity = 2
+
+    def as_oracle(self, compiler, connection):
+        if (isinstance(self.get_source_expressions()[0], Value) and
+                self.get_source_expressions()[0].value is None):
+            raise ValueError('Oracle does not allow the literal NULL for expression1.')
```

Comment:
Avoid calling `self.get_source_expressions()` twice and it becomes clearer:

```python
...
expression1 = self.get_source_expressions()[0]
if isinstance(expression1, Value) and expression1.value is None:
    raise ValueError('Oracle does not allow Value(None) for expression1.')
...
```

I've also tweaked the exception message to make it based on the argument provided in Python.